In [114]:
import pandas as pd

In [115]:
df = pd.read_csv("/content/drive/MyDrive/Datasets/superheroes_data.csv")

In [116]:
df.head(4)

id         name  intelligence  strength  speed  durability  power  combat  \
0   1       A-Bomb          38.0     100.0   17.0        80.0   24.0    64.0   
1   2   Abe Sapien          88.0      28.0   35.0        65.0  100.0    85.0   
2   3     Abin Sur          50.0      90.0   53.0        64.0   99.0    65.0   
3   4  Abomination          63.0      80.0   53.0        90.0   62.0    95.0   

                full-name            alter-egos  ...               race  \
0  Richard Milhouse Jones  No alter egos found.  ...              Human   
1          Abraham Sapien  No alter egos found.  ...      Icthyo Sapien   
2                     NaN  No alter egos found.  ...            Ungaran   
3            Emil Blonsky  No alter egos found.  ...  Human / Radiation   

              height                weight eye-color hair-color  \
0  ["6'8", '203 cm']  ['980 lb', '441 kg']    Yellow    No Hair   
1  ["6'3", '191 cm']   ['145 lb', '65 kg']      Blue    No Hair   
2  ["6'1", '185 cm']   ['200 lb', '90 kg']      Blue    No Hair   
3  ["6'8", '203 cm']  ['980 lb', '441 kg']     Green    No Hair   

                                          occupation    base  \
0  Musician, adventurer, author; formerly talk sh...       -   
1                            Paranormal Investigator       -   
2            Green Lantern, former history professor      Oa   
3                                             Ex-Spy  Mobile   

                                   group-affiliation  \
0  Hulk Family; Excelsior (sponsor), Avengers (ho...   
1         Bureau for Paranormal Research and Defense   
2           Green Lantern Corps, Black Lantern Corps   
3  former member of the crew of the Andromeda Sta...   

                                           relatives  \
0  Marlo Chandler-Jones (wife); Polly (aunt); Mrs...   
1                      Edith Howard (wife, deceased)   
2  Amon Sur (son), Arin Sur (sister), Thaal Sines...   
3            Nadia Dornova Blonsky (wife, separated)   

                                                 url  
0  https://www.superherodb.com/pictures2/portrait...  
1  https://www.superherodb.com/pictures2/portrait...  
2  https://www.superherodb.com/pictures2/portrait...  
3  https://www.superherodb.com/pictures2/portrait...  

[4 rows x 26 columns]

In [4]:
df.shape

(731, 26)

# Data set name :  Super Heros Dataset
# Rows :  731
# columns : 26
Column Descriptors:
id: A unique identifier for each character.

name: The superhero's alias or code name.

intelligence: A numerical representation of the character's intelligence level.

strength: A numerical value representing the character's physical strength.

speed: A numerical representation of how fast the character can move.

durability: A measure of the character’s resilience and ability to withstand damage.

power: A numerical value representing the character's overall power or abilities.

combat: A score depicting the character’s combat skills and experience.
full-name: The character’s real or full name, as opposed to their superhero alias.
alter-egos: Any other identities the character has used.
aliases: Alternate names or titles the character is known by.
place-of-birth: The location where the character was born or created.
first-appearance: The comic issue or media in which the character made their first appearance.
publisher: The company responsible for creating and publishing the character.
alignment: Whether the character is good, evil, or neutral.
gender: The gender of the character.
race: The character's species or race (e.g., human, mutant, alien).
height: The character’s height, often given in both feet and centimeters.
weight: The character’s weight, often provided in both pounds and kilograms.
eye-color: The color of the character’s eyes.
hair-color: The color of the character’s hair.
occupation: The character’s primary job or role, either before or alongside their superhero activities.
base: The main location where the character operates from.
group-affiliation: Teams, organizations, or alliances the character is part of.
relatives: Important family members related to the character.
url: A link to an image of the character or more detailed information.

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 731 entries, 0 to 730
Data columns (total 26 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 731 non-null    int64  
 1   name               731 non-null    object 
 2   intelligence       566 non-null    float64
 3   strength           629 non-null    float64
 4   speed              566 non-null    float64
 5   durability         566 non-null    float64
 6   power              566 non-null    float64
 7   combat             566 non-null    float64
 8   full-name          630 non-null    object 
 9   alter-egos         731 non-null    object 
 10  aliases            731 non-null    object 
 11  place-of-birth     731 non-null    object 
 12  first-appearance   731 non-null    object 
 13  publisher          716 non-null    object 
 14  alignment          731 non-null    object 
 15  gender             731 non-null    object 
 16  race               429 non

In [6]:
df.dtypes

id                     int64
name                  object
intelligence         float64
strength             float64
speed                float64
durability           float64
power                float64
combat               float64
full-name             object
alter-egos            object
aliases               object
place-of-birth        object
first-appearance      object
publisher             object
alignment             object
gender                object
race                  object
height                object
weight                object
eye-color             object
hair-color            object
occupation            object
base                  object
group-affiliation     object
relatives             object
url                   object
dtype: object

In [7]:
df['name'] = df['name'].astype('category')

In [10]:
df.dtypes

id                      int64
name                 category
intelligence          float64
strength              float64
speed                 float64
durability            float64
power                 float64
combat                float64
full-name              object
alter-egos             object
aliases                object
place-of-birth         object
first-appearance       object
publisher              object
alignment              object
gender                 object
race                   object
height                 object
weight                 object
eye-color              object
hair-color             object
occupation             object
base                   object
group-affiliation      object
relatives              object
url                    object
dtype: object

In [11]:
df['height']

0             ["6'8", '203 cm']
1             ["6'3", '191 cm']
2             ["6'1", '185 cm']
3             ["6'8", '203 cm']
4                 ['-', '0 cm']
                 ...           
726           ["5'5", '165 cm']
727    ['1000', '304.8 meters']
728            ["2'2", '66 cm']
729           ["5'7", '170 cm']
730           ["6'1", '185 cm']
Name: height, Length: 731, dtype: object

In [ ]:
list1 = ["6","120"]

In [33]:
import numpy as np

In [34]:
new_height = np.array([])

In [35]:
new_list = []

In [37]:
for i in df['height']:
  print(i.split(","))
  new_list.append(i.split(","))
  # print(i[9:12])
  #df[i] = df[i][9:12]

['["6\'8"', " '203 cm']"]
['["6\'3"', " '191 cm']"]
['["6\'1"', " '185 cm']"]
['["6\'8"', " '203 cm']"]
["['-'", " '0 cm']"]
['["6\'4"', " '193 cm']"]
["['-'", " '0 cm']"]
['["6\'1"', " '185 cm']"]
['["5\'8"', " '173 cm']"]
['["5\'10"', " '178 cm']"]
['["6\'3"', " '191 cm']"]
['["6\'2"', " '188 cm']"]
['["6\'4"', " '193 cm']"]
['["5\'11"', " '180 cm']"]
["['-'", " '0 cm']"]
["['-'", " '0 cm']"]
['["5\'10"', " '178 cm']"]
['["8\'0"', " '244 cm']"]
["['-'", " '0 cm']"]
['["8\'5"', " '257 cm']"]
['["6\'2"', " '188 cm']"]
["['-'", " '0 cm']"]
["['-'", " '0 cm']"]
['["6\'0"', " '183 cm']"]
['["5\'5"', " '165 cm']"]
['["5\'4"', " '163 cm']"]
["['-'", " '0 cm']"]
['["6\'0"', " '183 cm']"]
['["5\'11"', " '180 cm']"]
['["6\'11"', " '211 cm']"]
['["6\'0"', " '183 cm']"]
["['200'", " '61.0 meters']"]
["['-'", " '0 cm']"]
['["7\'6"', " '229 cm']"]
['["7\'0"', " '213 cm']"]
["['-'", " '0 cm']"]
['["5\'10"', " '178 cm']"]
['["6\'1"', " '185 cm']"]
['["5\'9"', " '175 cm']"]
['["6\'0"', " '183 cm']"]


In [39]:
for i in range(len(new_list)):
  new_list[i] = new_list[i][1]

In [40]:
new_list

[" '203 cm']",
 " '191 cm']",
 " '185 cm']",
 " '203 cm']",
 " '0 cm']",
 " '193 cm']",
 " '0 cm']",
 " '185 cm']",
 " '173 cm']",
 " '178 cm']",
 " '191 cm']",
 " '188 cm']",
 " '193 cm']",
 " '180 cm']",
 " '0 cm']",
 " '0 cm']",
 " '178 cm']",
 " '244 cm']",
 " '0 cm']",
 " '257 cm']",
 " '188 cm']",
 " '0 cm']",
 " '0 cm']",
 " '183 cm']",
 " '165 cm']",
 " '163 cm']",
 " '0 cm']",
 " '183 cm']",
 " '180 cm']",
 " '211 cm']",
 " '183 cm']",
 " '61.0 meters']",
 " '0 cm']",
 " '229 cm']",
 " '213 cm']",
 " '0 cm']",
 " '178 cm']",
 " '185 cm']",
 " '175 cm']",
 " '183 cm']",
 " '173 cm']",
 " '193 cm']",
 " '185 cm']",
 " '165 cm']",
 " '163 cm']",
 " '0 cm']",
 " '0 cm']",
 " '183 cm']",
 " '198 cm']",
 " '178 cm']",
 " '0 cm']",
 " '168 cm']",
 " '183 cm']",
 " '0 cm']",
 " '0 cm']",
 " '180 cm']",
 " '183 cm']",
 " '0 cm']",
 " '0 cm']",
 " '203 cm']",
 " '183 cm']",
 " '165 cm']",
 " '170 cm']",
 " '0 cm']",
 " '0 cm']",
 " '165 cm']",
 " '0 cm']",
 " '168 cm']",
 " '178 cm']",


In [117]:
new_height = []

In [118]:
for i in df['height']:

  print(i[9:12])


  if i[9:12] == "cm'":
    new_height.append("0")

  else:
    new_height.append(i[9:12])
  #df[i] = df[i][9:12]
  #df['height'][i] = i[9:12]



203
191
185
203
cm'
193
cm'
185
173
'17
191
188
193
'18
cm'
cm'
'17
244
cm'
257
188
cm'
cm'
183
165
163
cm'
183
'18
'21
183
61.
cm'
229
213
cm'
'17
185
175
183
173
193
185
165
163
cm'
cm'
183
198
'17
cm'
168
183
cm'
cm'
'18
183
cm'
cm'
203
183
165
170
cm'
cm'
165
cm'
168
'17
188
'17
198
'17
175
'18
173
cm'
cm'
201
cm'
188
cm'
165
cm'
cm'
'18
cm'
cm'
cm'
cm'
cm'
198
191
cm'
191
188
170
165
'17
cm'
cm'
183
185
170
188
183
170
'17
191
185
188
188
cm'
168
165
cm'
cm'
175
'17
218
cm'
 '3
cm'
cm'
183
cm'
183
cm'
cm'
165
cm'
196
cm'
193
cm'
198
170
183
193
157
183
170
cm'
cm'
203
175
cm'
183
188
193
198
cm'
188
cm'
'18
193
'18
175
cm'
cm'
cm'
185
173
cm'
175
170
201
cm'
175
cm'
'18
163
170
175
185
'17
183
cm'
cm'
226
'17
cm'
cm'
cm'
226
183
191
183
cm'
'18
168
cm'
cm'
198
cm'
191
'17
Hei
cm'
cm'
183
185
cm'
267
cm'
168
170
198
122
cm'
173
183
188
185
193
193
185
188
193
cm'
198
201
201
188
175
188
173
cm'
175
244
196
185
cm'
193
cm'
168
'18
175
cm'
185
 '1
168
cm'
cm'
168
193
188
191
183
196


In [120]:
new_height

['203',
 '191',
 '185',
 '203',
 '0',
 '193',
 '0',
 '185',
 '173',
 "'17",
 '191',
 '188',
 '193',
 "'18",
 '0',
 '0',
 "'17",
 '244',
 '0',
 '257',
 '188',
 '0',
 '0',
 '183',
 '165',
 '163',
 '0',
 '183',
 "'18",
 "'21",
 '183',
 '61.',
 '0',
 '229',
 '213',
 '0',
 "'17",
 '185',
 '175',
 '183',
 '173',
 '193',
 '185',
 '165',
 '163',
 '0',
 '0',
 '183',
 '198',
 "'17",
 '0',
 '168',
 '183',
 '0',
 '0',
 "'18",
 '183',
 '0',
 '0',
 '203',
 '183',
 '165',
 '170',
 '0',
 '0',
 '165',
 '0',
 '168',
 "'17",
 '188',
 "'17",
 '198',
 "'17",
 '175',
 "'18",
 '173',
 '0',
 '0',
 '201',
 '0',
 '188',
 '0',
 '165',
 '0',
 '0',
 "'18",
 '0',
 '0',
 '0',
 '0',
 '0',
 '198',
 '191',
 '0',
 '191',
 '188',
 '170',
 '165',
 "'17",
 '0',
 '0',
 '183',
 '185',
 '170',
 '188',
 '183',
 '170',
 "'17",
 '191',
 '185',
 '188',
 '188',
 '0',
 '168',
 '165',
 '0',
 '0',
 '175',
 "'17",
 '218',
 '0',
 " '3",
 '0',
 '0',
 '183',
 '0',
 '183',
 '0',
 '0',
 '165',
 '0',
 '196',
 '0',
 '193',
 '0',
 '198',
 '17

In [122]:
type(new_height[0])

str

In [126]:
for i in range(len(new_height)):
  print(new_height[i])
  if "'" in new_height[i]:
    print("yes")
    new_height[i] = new_height[i][1:]

203
191
185
203
0
193
0
185
173
'17
yes
191
188
193
'18
yes
0
0
'17
yes
244
0
257
188
0
0
183
165
163
0
183
'18
yes
'21
yes
183
61.
0
229
213
0
'17
yes
185
175
183
173
193
185
165
163
0
0
183
198
'17
yes
0
168
183
0
0
'18
yes
183
0
0
203
183
165
170
0
0
165
0
168
'17
yes
188
'17
yes
198
'17
yes
175
'18
yes
173
0
0
201
0
188
0
165
0
0
'18
yes
0
0
0
0
0
198
191
0
191
188
170
165
'17
yes
0
0
183
185
170
188
183
170
'17
yes
191
185
188
188
0
168
165
0
0
175
'17
yes
218
0
 '3
yes
0
0
183
0
183
0
0
165
0
196
0
193
0
198
170
183
193
157
183
170
0
0
203
175
0
183
188
193
198
0
188
0
'18
yes
193
'18
yes
175
0
0
0
185
173
0
175
170
201
0
175
0
'18
yes
163
170
175
185
'17
yes
183
0
0
226
'17
yes
0
0
0
226
183
191
183
0
'18
yes
168
0
0
198
0
191
'17
yes
Hei
0
0
183
185
0
267
0
168
170
198
122
0
173
183
188
185
193
193
185
188
193
0
198
201
201
188
175
188
173
0
175
244
196
185
0
193
0
168
'18
yes
175
0
185
 '1
yes
168
0
0
168
193
188
191
183
196
188
0
0
175
0
'97
yes
'16
yes
'19
yes
173
0
188
0
'1

In [127]:
new_height

['203',
 '191',
 '185',
 '203',
 '0',
 '193',
 '0',
 '185',
 '173',
 '17',
 '191',
 '188',
 '193',
 '18',
 '0',
 '0',
 '17',
 '244',
 '0',
 '257',
 '188',
 '0',
 '0',
 '183',
 '165',
 '163',
 '0',
 '183',
 '18',
 '21',
 '183',
 '61.',
 '0',
 '229',
 '213',
 '0',
 '17',
 '185',
 '175',
 '183',
 '173',
 '193',
 '185',
 '165',
 '163',
 '0',
 '0',
 '183',
 '198',
 '17',
 '0',
 '168',
 '183',
 '0',
 '0',
 '18',
 '183',
 '0',
 '0',
 '203',
 '183',
 '165',
 '170',
 '0',
 '0',
 '165',
 '0',
 '168',
 '17',
 '188',
 '17',
 '198',
 '17',
 '175',
 '18',
 '173',
 '0',
 '0',
 '201',
 '0',
 '188',
 '0',
 '165',
 '0',
 '0',
 '18',
 '0',
 '0',
 '0',
 '0',
 '0',
 '198',
 '191',
 '0',
 '191',
 '188',
 '170',
 '165',
 '17',
 '0',
 '0',
 '183',
 '185',
 '170',
 '188',
 '183',
 '170',
 '17',
 '191',
 '185',
 '188',
 '188',
 '0',
 '168',
 '165',
 '0',
 '0',
 '175',
 '17',
 '218',
 '0',
 "'3",
 '0',
 '0',
 '183',
 '0',
 '183',
 '0',
 '0',
 '165',
 '0',
 '196',
 '0',
 '193',
 '0',
 '198',
 '170',
 '183',
 '193

In [130]:
for i in range(len(new_height)):
  if len(new_height[i]) <3:
    new_height[i] = "0"

In [131]:
new_height

['203',
 '191',
 '185',
 '203',
 '0',
 '193',
 '0',
 '185',
 '173',
 '0',
 '191',
 '188',
 '193',
 '0',
 '0',
 '0',
 '0',
 '244',
 '0',
 '257',
 '188',
 '0',
 '0',
 '183',
 '165',
 '163',
 '0',
 '183',
 '0',
 '0',
 '183',
 '61.',
 '0',
 '229',
 '213',
 '0',
 '0',
 '185',
 '175',
 '183',
 '173',
 '193',
 '185',
 '165',
 '163',
 '0',
 '0',
 '183',
 '198',
 '0',
 '0',
 '168',
 '183',
 '0',
 '0',
 '0',
 '183',
 '0',
 '0',
 '203',
 '183',
 '165',
 '170',
 '0',
 '0',
 '165',
 '0',
 '168',
 '0',
 '188',
 '0',
 '198',
 '0',
 '175',
 '0',
 '173',
 '0',
 '0',
 '201',
 '0',
 '188',
 '0',
 '165',
 '0',
 '0',
 '0',
 '0',
 '0',
 '0',
 '0',
 '0',
 '198',
 '191',
 '0',
 '191',
 '188',
 '170',
 '165',
 '0',
 '0',
 '0',
 '183',
 '185',
 '170',
 '188',
 '183',
 '170',
 '0',
 '191',
 '185',
 '188',
 '188',
 '0',
 '168',
 '165',
 '0',
 '0',
 '175',
 '0',
 '218',
 '0',
 '0',
 '0',
 '0',
 '183',
 '0',
 '183',
 '0',
 '0',
 '165',
 '0',
 '196',
 '0',
 '193',
 '0',
 '198',
 '170',
 '183',
 '193',
 '157',
 '183'

In [136]:
for i in new_height:
  print(i)

203
191
185
203
0
193
0
185
173
0
191
188
193
0
0
0
0
244
0
257
188
0
0
183
165
163
0
183
0
0
183
61.
0
229
213
0
0
185
175
183
173
193
185
165
163
0
0
183
198
0
0
168
183
0
0
0
183
0
0
203
183
165
170
0
0
165
0
168
0
188
0
198
0
175
0
173
0
0
201
0
188
0
165
0
0
0
0
0
0
0
0
198
191
0
191
188
170
165
0
0
0
183
185
170
188
183
170
0
191
185
188
188
0
168
165
0
0
175
0
218
0
0
0
0
183
0
183
0
0
165
0
196
0
193
0
198
170
183
193
157
183
170
0
0
203
175
0
183
188
193
198
0
188
0
0
193
0
175
0
0
0
185
173
0
175
170
201
0
175
0
0
163
170
175
185
0
183
0
0
226
0
0
0
0
226
183
191
183
0
0
168
0
0
198
0
191
0
Hei
0
0
183
185
0
267
0
168
170
198
122
0
173
183
188
185
193
193
185
188
193
0
198
201
201
188
175
188
173
0
175
244
196
185
0
193
0
168
0
175
0
185
0
168
0
0
168
193
188
191
183
196
188
0
0
175
0
0
0
0
173
0
188
0
0
0
183
183
157
83 
142
0
0
0
0
185
183
0
0
0
185
188
0
0
0
62.
198
168
0
0
175
0
0
0
83 
198
0
0
170
188
0
0
83 
0
0
0
188
188
183
188
170
0
83 
185
191
165
175
185
175
0
170


In [137]:
print(type(new_height[0]))

<class 'str'>


In [143]:
for i in range(len(new_height)):
  print(new_height[i])
  if "Hei" == new_height[i]:
    new_height[i] = 0
  else:
    new_height[i] = float(new_height[i])

203.0
191.0
185.0
203.0
0.0
193.0
0.0
185.0
173.0
0.0
191.0
188.0
193.0
0.0
0.0
0.0
0.0
244.0
0.0
257.0
188.0
0.0
0.0
183.0
165.0
163.0
0.0
183.0
0.0
0.0
183.0
61.0
0.0
229.0
213.0
0.0
0.0
185.0
175.0
183.0
173.0
193.0
185.0
165.0
163.0
0.0
0.0
183.0
198.0
0.0
0.0
168.0
183.0
0.0
0.0
0.0
183.0
0.0
0.0
203.0
183.0
165.0
170.0
0.0
0.0
165.0
0.0
168.0
0.0
188.0
0.0
198.0
0.0
175.0
0.0
173.0
0.0
0.0
201.0
0.0
188.0
0.0
165.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
0.0
198.0
191.0
0.0
191.0
188.0
170.0
165.0
0.0
0.0
0.0
183.0
185.0
170.0
188.0
183.0
170.0
0.0
191.0
185.0
188.0
188.0
0.0
168.0
165.0
0.0
0.0
175.0
0.0
218.0
0.0
0.0
0.0
0.0
183.0
0.0
183.0
0.0
0.0
165.0
0.0
196.0
0.0
193.0
0.0
198.0
170.0
183.0
193.0
157.0
183.0
170.0
0.0
0.0
203.0
175.0
0.0
183.0
188.0
193.0
198.0
0.0
188.0
0.0
0.0
193.0
0.0
175.0
0.0
0.0
0.0
185.0
173.0
0.0
175.0
170.0
201.0
0.0
175.0
0.0
0.0
163.0
170.0
175.0
185.0
0.0
183.0
0.0
0.0
226.0
0.0
0.0
0.0
0.0
226.0
183.0
191.0
183.0
0.0
0.0
168.0
0.0
0.0
198.0
0.0
191.0
0.0

In [144]:
new_height = np.array(new_height)

In [145]:
new_height = pd.Series(new_height)

In [146]:
new_height = pd.DataFrame(new_height)

In [147]:
new_height

0
0    203.0
1    191.0
2    185.0
3    203.0
4      0.0
..     ...
726  165.0
727    0.0
728   66.0
729  170.0
730  185.0

[731 rows x 1 columns]

In [148]:
df.join(new_height)

id             name  intelligence  strength  speed  durability  power  \
0      1           A-Bomb          38.0     100.0   17.0        80.0   24.0   
1      2       Abe Sapien          88.0      28.0   35.0        65.0  100.0   
2      3         Abin Sur          50.0      90.0   53.0        64.0   99.0   
3      4      Abomination          63.0      80.0   53.0        90.0   62.0   
4      5          Abraxas          88.0      63.0   83.0       100.0  100.0   
..   ...              ...           ...       ...    ...         ...    ...   
726  727  Yellowjacket II          50.0      10.0   35.0        28.0   31.0   
727  728             Ymir          50.0     100.0   27.0       100.0   98.0   
728  729             Yoda          88.0      52.0   33.0        25.0  100.0   
729  730          Zatanna          81.0      10.0   23.0        28.0  100.0   
730  731             Zoom          50.0      10.0  100.0        28.0  100.0   

     combat               full-name            alter-egos  ...  \
0      64.0  Richard Milhouse Jones  No alter egos found.  ...   
1      85.0          Abraham Sapien  No alter egos found.  ...   
2      65.0                     NaN  No alter egos found.  ...   
3      95.0            Emil Blonsky  No alter egos found.  ...   
4      55.0                 Abraxas  No alter egos found.  ...   
..      ...                     ...                   ...  ...   
726    28.0             Rita DeMara  No alter egos found.  ...   
727    28.0                    Ymir  No alter egos found.  ...   
728    90.0                    Yoda  No alter egos found.  ...   
729    56.0          Zatanna Zatara  No alter egos found.  ...   
730    28.0          Hunter Zolomon  No alter egos found.  ...   

                       height                weight eye-color  \
0           ["6'8", '203 cm']  ['980 lb', '441 kg']    Yellow   
1           ["6'3", '191 cm']   ['145 lb', '65 kg']      Blue   
2           ["6'1", '185 cm']   ['200 lb', '90 kg']      Blue   
3           ["6'8", '203 cm']  ['980 lb', '441 kg']     Green   
4               ['-', '0 cm']      ['- lb', '0 kg']      Blue   
..                        ...                   ...       ...   
726         ["5'5", '165 cm']   ['115 lb', '52 kg']      Blue   
727  ['1000', '304.8 meters']      ['- lb', '0 kg']     White   
728          ["2'2", '66 cm']    ['38 lb', '17 kg']     Brown   
729         ["5'7", '170 cm']   ['127 lb', '57 kg']      Blue   
730         ["6'1", '185 cm']   ['181 lb', '81 kg']       Red   

           hair-color                                         occupation  \
0             No Hair  Musician, adventurer, author; formerly talk sh...   
1             No Hair                            Paranormal Investigator   
2             No Hair            Green Lantern, former history professor   
3             No Hair                                             Ex-Spy   
4               Black                              Dimensional destroyer   
..                ...                                                ...   
726  Strawberry Blond  Adventurer; former criminal, electronics engineer   
727           No Hair                                                  -   
728             White                                                  -   
729             Black                                                  -   
730             Brown                                                  -   

                      base                                  group-affiliation  \
0                        -  Hulk Family; Excelsior (sponsor), Avengers (ho...   
1                        -         Bureau for Paranormal Research and Defense   
2                       Oa           Green Lantern Corps, Black Lantern Corps   
3                   Mobile  former member of the crew of the Andromeda Sta...   
4                        -                                      Cosmic Beings   
..                     ...                                                ...   
726